src/DependencyInjection/Configuration.php

```diff
@@ -18,10 +18,23 @@ public function getConfigTreeBuilder(): TreeBuilder
         /** @var ArrayNodeDefinition $rootNode */
         $rootNode = $treeBuilder->getRootNode();
 
+        // @phpstan-ignore-next-line
         $rootNode->children()
-            ->scalarNode('shop_class')
-                ->defaultValue(AbstractShop::class)
+            ->scalarNode('storage')->defaultValue('doctrine')->end()
+            ->arrayNode('doctrine')
+                ->children()
+                    ->scalarNode('shop_class')
+                        ->defaultValue(AbstractShop::class)
+                    ->end()
                 ->end()
```

Comment:
Moving `shop_class` to `doctrine.shop_class` is a BC, isn't it?

Reply:
yes that's why i would release a new major version